haskell/cabal.bzl

```diff
@@ -552,6 +552,16 @@ Specify a fully qualified package name of the form <package>-<version>.
     return (dependencies, transitive_unpacked_sdists)
 
 def _stack_snapshot_impl(repository_ctx):
+
+    if repository_ctx.attr.snapshot and repository_ctx.attr.local_snapshot:
+        fail("Please specify either snapshot or repository_snapshot, not both")
+    elif repository_ctx.attr.snapshot:
+        snapshot = repository_ctx.attr.snapshot
+    elif repository_ctx.attr.local_snapshot:
+        snapshot = repository_ctx.path(repository_ctx.attr.local_snapshot)
+    else:
+        fail("Please specify one of snapshot or repository_snapshot")
```

Comment:
It doesn't look related to this PR, directly, but for some reason, errors in `stack_snapshot` seem to not stop the execution right away, and you get additional spurious errors. Example with a typo in the name of the snapshot:

```
ERROR: An error occurred during the fetch of repository 'stackage':
   Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: /home/aspiwack/projects/tweag/spot/master/backend/server/BUILD.bazel:15:1: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam and referenced by '//backend/server:server'
ERROR: Analysis of target '//backend/server:server' failed; build aborted: no such package '@stackage//': Not a regular file: /home/aspiwack/projects/tweag/spot/master/snapshot.yam
```